csrc/utils.h

```diff
@@ -386,28 +388,75 @@ std::string toDelimitedInlineString(
   return ss.str();
 }
 
-template <typename... Args>
 class DebugPrintScope {
  public:
+  template <typename... Args>
   DebugPrintScope(std::string name, Args... args) : name_(std::move(name)) {
     debug() << "Entering " << name_ << "("
             << toDelimitedString(std::forward_as_tuple(args...)) << ")"
             << std::endl;
   }
+
   ~DebugPrintScope() {
-    debug() << "Leaving " << name_ << std::endl;
+    debug() << "Leaving " << name_;
+    if (!return_.empty()) {
+      debug() << " returning " << return_;
+    }
+    if (!file_.empty()) {
+      debug() << " at " << file_;
+    }
+    if (line_ >= 0) {
+      debug() << ":" << line_;
+    }
+    debug() << std::endl;
+  }
+
+  template <typename T>
+  void setReturn(const T& ret, std::string file = "", int64_t line = -1) {
+    return_ = Printer<std::decay_t<T>>::toString(ret);
+    file_ = std::move(file);
+    line_ = line;
   }
 
  private:
   std::string name_;
```

Comment:
```suggestion
  // Typically the function name returned by `__func__`. 
  std::string name_;
```

Reply:
Is that right? Are parameters also included in `name_`?

Reply:
Yes, right. I added a comment, and besides that, I added `DEBUG_PRINT_SCOPE_NAME` which allows specifying a name. Sometimes `__func__` is not informative, for example, it can be `handle`. Args are not included in name.